### PR TITLE
[expo-updates][android][ios] allow reloading in dev mode in store clients

### PR DESCRIPTION
**Pull request by esamelson**
# Why

*less urgent fix than #10308 and #10309, review can wait if needed*

There was a small regression with the `Updates` JS module API in SDK 39, namely that it is no longer possible to use `reloadAsync` in development mode in the store clients. This PR fixes that behavior for the store clients.

# How

Add a separate `canRelaunch` check that the module will call, and always return `true` in the clients, rather than checking to see if expo-updates is being used to load the app. This works because we can always reload in the store clients even if expo-updates is not being used (e.g. in development mode).

# Test Plan

Test `Updates.reloadAsync()` in development sandbox in both UNVERSIONED and SDK 39.

### CODE REVIEW

File: packages/expo-updates/src/Updates.ts [hunk 1]
```diff
@@ -45,7 +45,7 @@ export async function reloadAsync(): Promise<void> {
   if (!ExpoUpdates.reload) {
     throw new UnavailabilityError('Updates', 'reloadAsync');
   }
-  if (__DEV__ || isUsingDeveloperTool) {
+  if (__DEV__ && !isUsingExpoDevelopmentClient) {
     throw new CodedError(
       'ERR_UPDATES_DISABLED',
       `You cannot use the Updates module in development mode. ${manualUpdatesInstructions}`
```
Comment (ide) [inline packages/expo-updates/src/Updates.ts:51]:
```suggestion
      `You cannot use the Updates module in development mode in a production app. ${manualUpdatesInstructions}`
```

**Review by ide — APPROVED** [review]
We'll need to cherry-pick this back and remember to rebuild the SDK 39 AAR.

**Comment by esamelson** [comment]
no need to rebuild the AAR as versioned expoview code is built from source, and this PR already backports the necessary changes. will land now and cherry-pick to sdk-39

**Comment by brentvatne** [comment]
https://github.com/expo/expo/issues/10464#issuecomment-703178030

**Comment by Slapbox** [comment]
@brentvatne we're still seeing `Updates.reloadAsync()` failing in development, saying it's not supported in development.

We're running SDK 39 and using `39.0.3.tar.gz`. Is there something I'm overlooking?

**Comment by brentvatne** [comment]
@Slapbox - probably you don't have expo-updates@0.3.5 in your project

**Comment by Slapbox** [comment]
@brentvatne you're right! Thank you, much appreciated!

Quick follow-up. We reference the [bundledNativeModules.json](https://github.com/expo/expo/blob/sdk-39/packages/expo/bundledNativeModules.json) for dependency versions by SDK version due to an issue with yarn/expo/monorepos where we [can't use `expo install [some package]`](https://github.com/expo/expo-cli/issues/1804#issuecomment-608645761)

**Comment by brentvatne** [comment]
@Slapbox - it's actually not out of date because it specifies a version range. we don't update it every time we release a patch version bump because we'd have too many releases of the expo package if we did that. if you ran `yarn upgrade` then `~0.3.3` would end up being updated to ~0.3.5

**Comment by Slapbox** [comment]
Ah good call. We usually just run `yarn`, not `yarn upgrade`. Thanks again.

**Pull request merged**